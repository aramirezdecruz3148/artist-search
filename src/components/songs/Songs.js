import React from 'react';
import PropTypes from 'prop-types';
import Song from './Song';

function Songs({ songId, songsArray }) {
  const songList = songsArray.map(({ songTitle }) => (
    <li key={songId}>
      <Song 
        songTitle={songTitle}
      />
    </li>
  ));

  return (
    <ul>
      {songList}
    </ul>
  );
}

Songs.propTypes = {
  songsArray: PropTypes.arrayOf(PropTypes.shape({
    songTitle: PropTypes.string.isRequired
  })).isRequired,
  songId: PropTypes.string.isRequired
};

export default Songs;
