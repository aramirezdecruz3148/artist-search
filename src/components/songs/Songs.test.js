import React from 'react';
import { shallow } from 'enzyme';
import Songs from './Songs';

describe('Songs component', () => {
  it('renders Songs', () => {
    const wrapper = shallow(<Songs songId='sdsd44234' songsArray={[{ songTitle: 'ekkjnw' }, { songTitle: '234234n' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
