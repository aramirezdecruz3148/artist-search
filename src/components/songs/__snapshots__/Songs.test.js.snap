// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs component renders Songs 1`] = `
<ul>
  <li
    key="sdsd44234"
  >
    <Song
      songTitle="ekkjnw"
    />
  </li>
  <li
    key="sdsd44234"
  >
    <Song
      songTitle="234234n"
    />
  </li>
</ul>
`;
